F/copy

vitiral opened the pull request:
- Adds a copy methods to copy a service or block from one instance to another
- copy methods have overwrite checking to ensure that accidental overwrite doesn't occur
- added unit tests to test new methods.
- `Service.copy` had to be added to facilitate copying.

Commit:
added copy_block and copy_service methods

Commit:
updated unit tests

hansmosh commented on pynio/service.py:
```diff
@@ -155,3 +155,13 @@ def delete(self):
     def pid(self):
         return self._status()['pid']
+
+    def copy(self, name):
```
Why you block.copy and service.copy? Wouldn't a user just make a new Service with some other service's config?

Service('new_service', config=service.config)

hansmosh commented:
I think these classes are starting to have too many specific methods if we add these. pynio just needs to provide the basic functionality so that another application using pynio can perform these sorts of things if they want.

Commit:
removed copy from block and service objects

Commit:
Updated add_block and add_service to allow copying

broke one unit test because you are now required to do
block=instance.add_block(block)

removed the copy_service and copy_block methods.

hansmosh commented:
Looking good!
